core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -238,8 +238,8 @@ public synchronized void connect() throws AlluxioStatusException {
             getServiceName(), mAddress);
         return;
       } catch (IOException e) {
-        LOG.debug("Failed to connect ({}) with {} @ {}: {}", retryPolicy.getAttemptCount(),
-            getServiceName(), mAddress, e.getMessage());
+        LOG.debug("Failed to connect ({}) with {} @ {}", retryPolicy.getAttemptCount(),
+            getServiceName(), mAddress, e);
```

Comment:
I vaguely remember @bf8086 discovering this doesn't print the error as expected. Or was that something else?

Reply:
I have an issue created on this error message: https://github.com/Alluxio/alluxio/issues/11197
It shows very confusing messages, like
```
$ ./bin/alluxio fs mount /mnt/s3 s3://analytics-zoo-data
Failed to connect to FileSystemMasterClient @ localhost:19998 after 8 attempts
```
Let me improve the message here too

Reply:
IIRC the issue was more about we lose important information when reporting a server error. It was not related to the server log.